keras/backend/tensorflow_backend.py

```diff
@@ -1924,3 +1925,24 @@ def ctc_decode(y_pred, input_length, greedy=True, beam_width=100,
                      for st in decoded]
 
     return (decoded_dense, log_prob)
+
+
+class ReverseGradient(object):
```

Comment:
Another option: generate a unique gradient name with e.g. system time and forget `num_calls` altogether.